x-pack/plugins/ml/server/models/job_validation/validate_model_memory_limit.test.ts

```diff
@@ -61,29 +63,46 @@ describe('ML - validateModelMemoryLimit', () => {
     },
   };
 
+  // mock estimate model memory
+  const modelMemoryEstimateResponse: ModelMemoryEstimate = {
+    model_memory_estimate: '40mb',
+  };
+
+  interface MockAPICallResponse {
+    'ml.estimateModelMemory'?: ModelMemoryEstimate;
+  }
+
   // mock callWithRequest
   // used in three places:
   // - to retrieve the info endpoint
   // - to search for cardinality of split field
   // - to retrieve field capabilities used in search for split field cardinality
-  function callWithRequest(call) {
-    if (typeof call === undefined) {
-      return Promise.reject();
-    }
-
-    let response = {};
-    if (call === 'ml.info') {
-      response = mlInfoResponse;
-    } else if (call === 'search') {
-      response = cardinalitySearchResponse;
-    } else if (call === 'fieldCaps') {
-      response = fieldCapsResponse;
-    }
-    return Promise.resolve(response);
-  }
-
-  function getJobConfig(influencers = [], detectors = []) {
-    return {
+  const getMockCallWithRequest = ({
+    'ml.estimateModelMemory': estimateModelMemory,
+  }: MockAPICallResponse = {}) =>
+    ((call: string) => {
+      if (typeof call === undefined) {
+        return Promise.reject();
+      }
+
+      let response = {};
+      if (call === 'ml.info') {
+        response = mlInfoResponse;
+      } else if (call === 'search') {
+        response = cardinalitySearchResponse;
+      } else if (call === 'fieldCaps') {
+        response = fieldCapsResponse;
+      } else if (call === 'ml.estimateModelMemory') {
+        response = estimateModelMemory || modelMemoryEstimateResponse;
+      }
+      return Promise.resolve(response);
+    }) as APICaller;
+
+  function getJobConfig(
+    influencers: string[] = [],
+    detectors: AnomalyDetectionJob['analysis_config']['detectors'] = []
```

Comment:
this could be a `Detector []`

Reply:
Changed in https://github.com/elastic/kibana/pull/60376/commits/0ec72638475bfebead195972a3034f6012011104